proxyserver/objhandlers.go

```diff
@@ -194,7 +194,7 @@ func (server *ProxyServer) ObjectPutHandler(writer http.ResponseWriter, request
 			return
 		}
 	}
-	if request.Header.Get("Content-Type") == "" {
+	if request.Header.Get("Content-Type") == "" || common.LooksTrue(request.Header.Get("X-Detect-Content-Type")) {
```

Comment:
Seems weird to let people both specify a content-type AND x-detect-content-type...